RiotSwiftUI/Modules/Spaces/AddRoomSelector/Coordinator/AddRoomSelectorViewProvider.swift

```diff
@@ -14,10 +14,11 @@
 // limitations under the License.
 //
 
-import Foundation
+import SwiftUI
 
-/// Actions to be performed on the `ViewModel` State
-enum SpaceCreationMatrixItemListStateAction {
-    case updateItems([SpaceCreationMatrixItem])
-    case updateSelection(Set<String>)
+class AddRoomSelectorViewProvider: MatrixItemChooserCoordinatorViewProvider {
+    @available(iOS 14, *)
+    func view(with viewModel: MatrixItemChooserViewModelType.Context) -> AnyView {
```

Comment:
Does it work if this function (and presumably in `MatrixItemChooserCoordinatorViewProvider` etc) returns `some View` instead of `AnyView`? I *think* that generally we want to avoid the use of `AnyView` where we can.

Reply:
The problem here comes from the `MatrixItemChooserCoordinatorViewProvider` protocol. Instead of `AnyView` I can use an associated type as below:
```
@available(iOS 14.0, *)
internal protocol MatrixItemChooserCoordinatorViewProvider {
    associatedtype SomeView: View
    func view(with viewModel: MatrixItemChooserViewModelType.Context) -> SomeView
}
```

But if I do so I have the following error in `MatrixItemChooserCoordinatorParameters`:
```
Protocol 'MatrixItemChooserCoordinatorViewProvider' can only be used as a generic constraint because it has Self or associated type requirements
```

As of now, I didn't find a solution for this error other than using `AnyView` but you may have an idea for that.

Reply:
Well that's sad. Nope I can't see anything else that would work here either. I guess type erasing to `AnyView` is the only option then 😕